lib/room.cpp

```diff
@@ -1269,6 +1269,16 @@ const EventPtr& Room::accountData(const QString& type) const
     return it != d->accountData.end() ? it->second : NoEventPtr;
 }
 
+QStringList Room::accountDataEventTypes() const
+{
+    QStringList events;
+    events.reserve(d->accountData.size());
+    for (const auto &[key, value] : d->accountData) {
```

Comment:
To avoid unneeded COW-copying (it's a shame `const` doesn't automatically propagate inside pimpls...):
```suggestion
    for (const auto& [key, value] : std::as_const(d->accountData)) {
```